### PR TITLE
Feat/master/autogeneration

### PR DESCRIPTION
**Pull request by ognjenkatic**
This PR is a major refactor of the client project. It switches over from hand written client code to semi auto generated (NSwag) to achieve full support for conductor API and make it easier to follow changes. There is a layer of abstraction in between the caller and the auto generated code that makes the method names a bit more meaningful.

We also took the opportunity to move to .Net6 and do some code changes to take advantage of some of the newer language features.

### CODE REVIEW

**Review by boma96 — CHANGES_REQUESTED** [review]
Some comments to consider

File: src/ConductorSharp.Client/ConductorSharp.Client.csproj [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
+		<ImplicitUsings>enable</ImplicitUsings>
+		<Nullable>enable</Nullable>
 	</PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="7.0.1" />
+    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="8.0.0" />
     <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
-    <PackageReference Include="System.Text.Json" Version="7.0.3" />
+    <PackageReference Include="System.Text.Json" Version="8.0.0" />
```
[the remark this answers is not in the document]
Comment (ognjenkatic) [reply src/ConductorSharp.Client/ConductorSharp.Client.csproj]: Nice catch, i believe we needed this before we migrated the project over to .net6, there was something missing in .netcore. Removed it now.

File: src/ConductorSharp.Engine/ExecutionManager.cs [hunk 2]
```diff
@@ -149,13 +176,20 @@ private async Task PollAndHandle(TaskToWorker scheduledWorker, CancellationToken
...
+                        Status = TaskResultStatus.FAILED,
+                        ReasonForIncompletion = exception.Message,
+                        OutputData = SerializationHelper.ObjectToDictionary(errorMessage, ConductorConstants.IoJsonSerializerSettings),
+                        WorkflowInstanceId = pollResponse.WorkflowInstanceId,
+                    },
+                    cancellationToken
                 );
+
+                await _taskManager.LogAsync(pollResponse.TaskId, exception.Message, cancellationToken);
+                await _taskManager.LogAsync(pollResponse.TaskId, exception.StackTrace, cancellationToken);
```
[the remark this answers is not in the document]
Comment (ognjenkatic) [reply src/ConductorSharp.Engine/ExecutionManager.cs]: Added, however, note the comment i added above the calls.

**Comment by ognjenkatic** [comment]
> @ognjenkatic Also can you put nswag file used to generate models and client into repository

This would be two files, as we split it into models and client file. Lets see if we like what we made and do some testing. If it works out, I will add them to the .sourcegen directory.

**Review by boma96 — APPROVED** [review]
We should test this parallel scenario before making stable release. In following days I plan to introduce some changes so after that we can do proper release

**Review by boma96 — CHANGES_REQUESTED** [review]
@ognjenkatic I just noticed that you used WorkflowTaskType instead of Type in Builders.
According to [docs](https://orkes.io/content/reference-docs/worker-task) `type` should be used. Also swagger file contains both `type` and `workflowTaskType` for some reason

**Comment by ognjenkatic** [comment]
> @ognjenkatic I just noticed that you used WorkflowTaskType instead of Type in Builders. According to [docs](https://orkes.io/content/reference-docs/worker-task) `type` should be used. Also swagger file contains both `type` and `workflowTaskType` for some reason.

Conductor seems to allow setting workflow task type in two ways, either as a string `type` or enum `workflowTaskType`.

In the generated models `setWorkflowTaskType` is a non nullable enum, so if not set to actual task type explicitly on our end it will have the first enum value `SIMPLE`.
image

**Comment by ognjenkatic** [comment]
Both string and enum values are now sent for type.

**Pull request merged**